### PR TITLE
Multiple fixes

### LINKED ISSUE #204: Unclear parameter descriptions for Show-InstallationWelcome
There are a couple of unclear descriptions for the parameters of Show-InstallationWelcome. Since this is a complex function with many parameters it's important for these to be very precise.

PersistPrompt - 'Specify whether to make the prompt persist...'. It is unclear which prompt. Since the switch comes immediately after 'PromptToSave', is this the prompt being referenced?

BlockExecution - '... the process/application...'. It should be made clear that these are the processes referred to in 'CloseApps'. Also, I assume that this is ignored if there is no 'CloseApps' specified.

AllowDefer and AllowDeferCloseApps - Are these meant to be mutually exclusive? What happens if you specify both?

### LINKED ISSUE #610: Show-InstallationProgress Functionality
We are moving our default template from 3.8.2 to 3.8.4 and something has changed in how Show-InstallationProgress functions. In the past when you launched this function the windowlocation was TopCenter, but since 3.8.3 the default location is in now dead center of the screen. 

If you use the -WindowLocation TopCenter parameter it observes the previous default behaviour. The function parameter information even states the default is top centers.

.PARAMETER WindowLocation
	The location of the progress window. Default: just below top, centered.

Is center of screen supposed to be the new behaviour, if not is there an easy fix to ensure top center is observed as default.I would assume the behaviour should be topcenter (default), bottomright  and center should be the options, but I could be wrong.

Any assistance is greatly appreciated.

### LINKED ISSUE #612: Error when running AppDeployToolkitMain.ps1 on Domain Controller
**Describe the bug**
On Domain Controllers there is no group "LocalPowerUsersGroup", so line 270 ([string]$LocalPowerUsersGroup = (New-Object -TypeName System.Security.Principal.SecurityIdentifier -ArgumentList ([Security.Principal.WellKnownSidType]::'BuiltinPowerUsersSid', $null)).Translate([System.Security.Principal.NTAccount]).Value) fails. 

**To Reproduce**
Run AppDeployToolkitMain.ps1 interactively on a Domain Controller

**Screenshots**
---

**Toolkit Version:**
3.8.4

**Powershell Version:**
5.1

**Additional context**
Occurs only on domain controllers, on workstations and member servers everything's fine.

### LINKED ISSUE #614: Active Setup doesn't executes the StubPath file for the current user ONLY IF the Package is started under SYSTEM with ServiceUIx64.exe
It worked under Ver. 3.8.3, but there i reported the bug #579 that the Active Setup is executed every time you start the installation a second time. This is solved, but now the user has to log off and on that the Active Setup runs.

We start our packages under SYSTEM with ServiceUIx64.exe because we want to display PSADT-Prompts.

Example:
Start a CMD as Admin => psexec -s cmd => now the cmd runs under SYSTEM =>
ServiceUIx64.exe -Process:explorer.exe Deploy-Application.exe "-DeployMode 'Interactive' -DeploymentType 'Install' -AllowRebootPassThru

---
Logfile:
<![LOG[[Pre-Installation] :: Adding Active Setup Key for local machine: [Registry::HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Active Setup\Installed Components\Mozilla_Thunderbird_78.7.0_x86_DE_01].]LOG]!><time="22:17:34.87560" date="02-02-2021" component="Set-ActiveSetup" context="NT-AUTORITÄT\SYSTEM" type="1" thread="5248" file="Deploy-Application.ps1">
<![LOG[[Pre-Installation] :: Create registry key [Registry::HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Active Setup\Installed Components\Mozilla_Thunderbird_78.7.0_x86_DE_01].]LOG]!><time="22:17:34.90660" date="02-02-2021" component="Set-RegistryKey" context="NT-AUTORITÄT\SYSTEM" type="1" thread="5248" file="Deploy-Application.ps1">
<![LOG[[Pre-Installation] :: Set registry key value: [Registry::HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Active Setup\Installed Components\Mozilla_Thunderbird_78.7.0_x86_DE_01] [(Default) = Thunderbird Installation. Bitte warten...].]LOG]!><time="22:17:34.90660" date="02-02-2021" component="Set-RegistryKey" context="NT-AUTORITÄT\SYSTEM" type="1" thread="5248" file="Deploy-Application.ps1">
<![LOG[[Pre-Installation] :: Set registry key value: [Registry::HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Active Setup\Installed Components\Mozilla_Thunderbird_78.7.0_x86_DE_01] [StubPath = C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -ExecutionPolicy Bypass -NoProfile -NoLogo -WindowStyle Hidden -Command "& { & \"C:\Program Files (x86)\Mozilla Thunderbird\ActiveSetupThunderbird.ps1\"}"].]LOG]!><time="22:17:34.92260" date="02-02-2021" component="Set-RegistryKey" context="NT-AUTORITÄT\SYSTEM" type="1" thread="5248" file="Deploy-Application.ps1">
<![LOG[[Pre-Installation] :: Set registry key value: [Registry::HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Active Setup\Installed Components\Mozilla_Thunderbird_78.7.0_x86_DE_01] [Version = 78,7,0,1.0].]LOG]!><time="22:17:34.93760" date="02-02-2021" component="Set-RegistryKey" context="NT-AUTORITÄT\SYSTEM" type="1" thread="5248" file="Deploy-Application.ps1">
<![LOG[[Pre-Installation] :: Set registry key value: [Registry::HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Active Setup\Installed Components\Mozilla_Thunderbird_78.7.0_x86_DE_01] [IsInstalled = 1].]LOG]!><time="22:17:34.95360" date="02-02-2021" component="Set-RegistryKey" context="NT-AUTORITÄT\SYSTEM" type="1" thread="5248" file="Deploy-Application.ps1">
<![LOG[[Pre-Installation] :: Failed to set Active Setup registry entry. 
Error Record:
-------------In D:\3.8.4\AppDeployToolkit\AppDeployToolkitMain.ps1:10768 Zeichen:58
+ ... et-RegistryKey -Key $HKCUActiveSetupKey -SID $UserProfile.SID -Value  ...
+                                                  ~~~~~~~~~~~~~~~~
Error Inner Exception(s):
-------------------------]LOG]!><time="22:17:35.03160" date="02-02-2021" component="Set-ActiveSetup" context="NT-AUTORITÄT\SYSTEM" type="3" thread="5248" file="Deploy-Application.ps1">
<![LOG[[Installation] :: [D:\3.8.4\Files\Thunderbird Setup 78.7.0.exe] is a valid fully qualified path, continue.]LOG]!><time="22:17:35.04760" date="02-02-2021" component="Execute-Process" context="NT-AUTORITÄT\SYSTEM" type="1" thread="5248" file="Deploy-Application.ps1">
---

I would prefer getting back the behaviour from 3.8.3 and use my workaroud: In my "ActiveSetupThunderbird.ps1" i check the "Version" Regkey from HKLM and HKCU. If it is the same, i quit my ps1.
Sorry for the inconvenience.

### PR DESCRIPTION
**Pull request by luki1412**
- Fixed a bug in the function Show-InstallationRestartPrompt, that caused incorrect handling of parameters when running the function asynchronously
- Fixed help text for some functions that contained old/incorrect information
- Fixed $LocalPowerUsersGroup throwing errors on machines where it does not exist (e.g. Domain Controllers)
- Changed function Set-ActiveSetup not executing the process for the current user correctly
  - Fixed the function not executing the process for the current user correctly
  - The function now also checks Version and IsInstalled properties when attempting to run the process for current user
  - Added more logging messages so it is clear from the logs why it was or was not executed for the current user
  - IsInstalled property is no longer added to the HKCU registry entry
  - DisableActiveSetup now also disables ExecuteForCurrentUser
  - Extended function help text
- Fixed Resolve-Error not returning full error record information
- Adds a changelog entry for #609 

Fixes #612
Fixes #610 
Fixes #204
Fixes #614 

Before submitting this Pull Request, I made sure:

- [X] I tested the toolkit with my changes and made sure it doesn't break other code.

- [X] I updated the documentation with the changes I made.

- [X] The code I changed has comments with explanation.

- [X] The encoding of the file wasn't changed. It is still UTF8 with BOM.

### CODE REVIEW

File: CHANGELOG.txt [hunk 1]
```diff
@@ -1,4 +1,9 @@
-﻿**Version 3.8.4 [26/01/2021]**
+﻿**Version 3.8.5 [??/??/2021]**
+- Fixed a bug in the function Show-InstallationRestartPrompt, that caused incorrect handling of parameters when running the function asynchronously
+- Fixed help text for some functions that contained old/incorrect information
+- Fixed $LocalPowerUsersGroup throwing errors on machines where it does not exist
```
Comment (sintaxasn) [inline CHANGELOG.txt]: Needs to have "(e.g. Domain Controllers) at the end of the line. If a customer encounters the issue, we should make sure they find this issue text and upgraded to latest.

File: Toolkit/AppDeployToolkit/AppDeployToolkitMain.ps1 [hunk 2]
```diff
@@ -266,9 +266,11 @@ Else {
...
-[string]$LocalAdministratorsGroup = (New-Object -TypeName System.Security.Principal.SecurityIdentifier -ArgumentList ([Security.Principal.WellKnownSidType]::'BuiltinAdministratorsSid', $null)).Translate([System.Security.Principal.NTAccount]).Value
+[string]$LocalUsersGroup = (New-Object -TypeName 'System.Security.Principal.SecurityIdentifier' -ArgumentList ([Security.Principal.WellKnownSidType]::'BuiltinUsersSid', $null)).Translate([System.Security.Principal.NTAccount]).Value
+try {
+	[string]$LocalPowerUsersGroup = (New-Object -TypeName 'System.Security.Principal.SecurityIdentifier' -ArgumentList ([Security.Principal.WellKnownSidType]::'BuiltinPowerUsersSid', $null)).Translate([System.Security.Principal.NTAccount]).Value
+} catch { }
+[string]$LocalAdministratorsGroup = (New-Object -TypeName 'System.Security.Principal.SecurityIdentifier' -ArgumentList ([Security.Principal.WellKnownSidType]::'BuiltinAdministratorsSid', $null)).Translate([System.Security.Principal.NTAccount]).Value
```
Comment (sintaxasn) [inline Toolkit/AppDeployToolkit/AppDeployToolkitMain.ps1]: For consistency, can we reuse a scriptblock or private function to ensure we either get a valid result or return a $null? Something like this 
```suggestion
$GetAccountNameUsingSid = [scriptblock]{ 
	param (
		[string] $SecurityIdentifier = $null
		)
	
	Try {
		[string] $result = (New-Object -TypeName System.Security.Principal.SecurityIdentifier -ArgumentList ([Security.Principal.WellKnownSidType]::"$SecurityIdentifier", $null)).Translate([System.Security.Principal.NTAccount]).Value
		return $result
	}
	Catch { 
		Return $null 
	}
	
$LocalSytemNTAccount = & $GetAccountNameUsingSid  'LocalSystemSid'
$LocalUsersGroup = & $GetAccountNameUsingSid 'BuiltinUsersSid'
$LocalPowerUsersGroup = & $GetAccountNameUsingSid  'BuiltinPowerUsersSid'
$LocalAdministratorsGroup = & $GetAccountNameUsingSid 'BuiltinAdministratorsSid'
```

**Review by sintaxasn — CHANGES_REQUESTED** [review]
Can you take a look at my conmments?

File: Toolkit/AppDeployToolkit/AppDeployToolkitMain.ps1 (hunk 2 above)
Comment (luki1412) [reply Toolkit/AppDeployToolkit/AppDeployToolkitMain.ps1]: Done

File: CHANGELOG.txt (hunk 1 above)
Comment (luki1412) [reply CHANGELOG.txt]: Done

**Review by sintaxasn — APPROVED** [review]

**Review by sintaxasn — APPROVED** [review]
Let me know if this is good to merge

**Comment by luki1412** [comment]
New one looks like this. The Message/ErrorID one splits in two if they are different.
new

**Pull request merged**